invenio_app_rdm/records_ui/views/filters.py

```diff
@@ -121,3 +122,24 @@ def get_scheme_label(scheme):
     scheme_to_label = current_app.config.get("RDM_RECORDS_IDENTIFIERS_SCHEMES", {})
 
     return scheme_to_label.get(scheme, {}).get("label", scheme)
+
+
+def localize_number(value):
+    locale_value = current_app.config.get("BABEL_DEFAULT_LOCALE")
+    number = int(value)
+    return format_decimal(number, locale=locale_value)
+
+def compact_number(value):
+    locale_value = current_app.config.get("BABEL_DEFAULT_LOCALE")
+    number = int(value)
+    decimals = 0
+
+    if number > 1000000:
```

Comment:
```suggestion
    if number > 1_000_000:
```

minor: to avoid the "magic number" we could make this an optional parameter to the Jinja filter, so that someone overriding the template could change this.

Ditto for other places where this is hardcoded.

(also better Python number formatting :) )